### PR TITLE
Ensure consistent return of dandisets with ID zero

### PR DESCRIPTION
**Pull request by AlmightyYakob**
Re-fixes issue addressed by #39. Somehow that change got taken out of master.

### CODE REVIEW

File: publish/models/dandiset.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ class Meta:
 
     @property
     def identifier(self) -> Optional[str]:
-        return f'{self.id:06}' if self.id else ''
+        return f'{self.id:06}' if self.id is not None else ''
```
Comment (mgrauer) [inline publish/models/dandiset.py:33]:
```suggestion
        return f'{self.id:06}' if self.id is not None else '' # Guard the case of identifier 0
```

Comment (AlmightyYakob) [reply publish/models/dandiset.py:33]: That comment only really makes sense in the context of the diff here. Would it make more sense to generally describe the function? Something like

```# Return the zero-padded id if it exists.```

Comment (brianhelba) [reply publish/models/dandiset.py:33]: `# Return the zero-padded id if it exists.` describes what the code does (which any developer can already understand), not *why*. It's fine for a docstring, but unhelpful as a code comment.

If we feel a comment is necessary, perhaps just be explicit: `# Compare against None, to allow id 0`.

Comment (brianhelba) [reply publish/models/dandiset.py:33]: Also, it's our style convention to have code comments always occupy their own line, unless it's a `# noqa: `.

**Pull request merged**